imperial_coldfront_plugin/oidc.py

```diff
@@ -53,4 +57,14 @@ def get_userinfo(
         """
         user_info = super().get_userinfo(access_token, id_token, payload)
         user_info["preferred_username"] = payload["preferred_username"]
+
+        # get user uid from Microsoft Graph API using the access token
+        # uid is stored under a custom attribute
+        response = requests.get(
+            "https://graph.microsoft.com/v1.0/me?$select=onPremisesExtensionAttributes",
```

Comment:
Should this URL be defined in the settings (eventually)?

Reply:
It's not a setting per se in that it we wouldn't expect it to vary but it probably shouldn't be such a magic value.